### ISSUE TITLE
aws_snapshot data source

### ISSUE THREAD

**Issue by kojiromike** [hashicorp/terraform#8828]
It would be really nice to have a data source to provide snapshot ids when provisioning volumes.
### Terraform Version

`Terraform v0.7.3`
### Affected Resource(s)

Does not exist yet. This is a feature request.
### Terraform Configuration Files (conceptual)

``` hcl
data "aws_ebs_snapshot" "example" {
  most_recent = true
  filter {
    name = "name"
    values = ["xmpl-snap-vpc-*"]
  }
  owners = ["self"]
}
resource "aws_ebs_volume" "example" {
    availability_zone = "us-west-2a"
    snapshot_id = "${data.aws_snapshot.example.id}"
    tags {
        Name = "HelloWorld"
    }
}
```
### Desired Behavior

This would pick the most recent snapshot matching the filters given, and use that to create the described volume.
### References

This is similar to the aws_ami data source and how it's used to create instances.

**Comment by apparentlymart** [comment]
Once #6819 is merged we'll have a helper function to ease the creation of new data sources for EC2 objects, at which point it shouldn't be hard to add one for EBS snapshots.

Since we have an `aws_ebs_volume` resource already I think we should call this data source `aws_ebs_snapshot`.

**Comment by kojiromike** [comment]
@apparentlymart Good thought on the name. I updated the request.

**Comment by flypenguin** [comment]
any chance for TF 0.7.10 (or whatever comes after 0.7.9)? would be super-awesome.

**Comment by stack72** [comment]
Hi @flypenguin 

This didn't make 0.7.10 - but it will hit soon ™️

**Comment by kojiromike** [comment]
Thank you! Since this page is higher ranking than the docs on Google for the query I used, putting a link here:

https://www.terraform.io/docs/providers/aws/d/ebs_snapshot.html
